### DISCUSSION TITLE
Constant loop when binding to array and modifying it

### DISCUSSION THREAD

**Discussion by AndriusPreibys** [alpinejs/alpine#2425] [7. Bugs (archived)] [answered]
Using Alpine v2.8.2

The goal is to bind checkboxes to array (with x-model - works fine), and disable/untick checkboxes depending on another checkbox.
You can see the example below: https://codepen.io/guzas/pen/vYeYjxP

Ticking first checkbox logs "condition bool" in console.
Unticking first, and ticking the second logs many "condition array" in console.

Why does Alpine loops the second function indefinitely when using array instead of boolean? How to prevent such behavior?
This works properly in V3 but still have to support IE 11.

**Comment by SimoTod** [comment, marked as answer]
in your code, you check the condition and when it's not false, you do
```javascript
if (condition) {
  this.valAsArray = [];
  console.log('condition array');
}
```
in v2, this triggers a full refresh ([] is a new object so it's seen as a change) which re-evaluates the condition and so on.

V3 is a bit better at understanding what has changed and what needs refreshing.

For v2, you need something like https://codepen.io/SimoTod/pen/wvrvQRZ

**Reply by AndriusPreibys** [reply]
Thanks!
